src/App.tsx

```diff
@@ -1,26 +1,52 @@
-import React from 'react';
-import logo from './logo.svg';
+import { useState, useEffect } from 'react';
 import './App.css';
+import './normalize.css'
+import { Wine } from './interfaces/wine_interface';
+import { WineList } from './components/WinesList/WineList'
+import { fetchAllWines } from './api/wine_api'
 
-function App() {
+const App = () => {
+  const [wines, setWines] = useState<Wine[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState("")
+
+  useEffect(() => {
+    
+    (async () => {try {
+      const wines = await fetchAllWines()
+      setWines(wines)
+      setIsLoading(false)
+    } catch (error) {
+      setError("Unable to fetch data, please try again later!")
+      setWines([])
+      setIsLoading(false)
+      }
+    })()
+
+    return () => {
+      setWines([])
+      setIsLoading(true)
+      setError("")
+    }
+  }, [])
+
+  const renderComponentOrError = () => {
+    if (error) {
+      return <h2 data-testid='error' className='error-style'>{error}</h2>
+    }
+    return <WineList winesList={wines} />
+  }
+  
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1 className="title-welcome">Welcome to Wine-O Inventory</h1>
+      {isLoading ? (
+        <h1 className='loading'>Loading...</h1>
```

Comment:
This seems like a reusable component, in fact I'd expect MUI to have a loading component you can use instead. Generally, in React we try to abstract away styles. i.e. I might expect App.tsx to return something like this instead. ```
<Container>
  <Title>...</Title>
  {isLoading ? ( <Loading />) :  ( <WineList wines={wines} error={error}/ >}
</Container>
```
Whether we actually want WineList to know about errors is a different question, but this is a more idiomatic example of what you have above.

Reply:
Another to thing checkout here re: loading, is Suspense it might still be experimental, but I think it's only a matter of time before it becomes part of a React release: https://reactjs.org/docs/concurrent-mode-suspense.html

Reply:
I'll go ahead and abstract away errors and loading into their own components, and rewrite the logic in a more idiomatic way. 

That's a really cool feature, and worth testing out. Should I test it out in this story or try to incorporate it later down the road?

Reply:
Whatever you want. I think it's nice to use the apprenticeship to mess around and try out new things since you aren't writing production code that folks are depending on, but up to you.